src/composables/transferSendHandler.ts

```diff
@@ -0,0 +1,52 @@
+import { computed } from 'vue';
+import { useRoute } from 'vue-router';
+
+import { TransferFormModel, WalletRouteMeta } from '@/types';
+import {
+  IS_EXTENSION,
+  MODAL_TRANSFER_SEND,
+  TRANSFER_SEND_DATA_LOCAL_STORAGE_KEY,
+} from '@/constants';
+import { useModals } from '@/composables';
+import { getLocalStorageItem, removeLocalStorageItem, setLocalStorageItem } from '@/utils';
+
+export function useTransferSendHandler() {
+  const { meta } = useRoute();
+  const { openModal } = useModals();
+
+  const routeMeta = computed<WalletRouteMeta | undefined>(() => meta);
+
+  /**
+  * Save form data on local storage to restore it
+  * e.g. after extension closes & opens window to ask camera permission
+  */
+  function save(formModel: TransferFormModel) {
+    setLocalStorageItem([TRANSFER_SEND_DATA_LOCAL_STORAGE_KEY], formModel);
+  }
+
+  function restore() {
+    const transferSendData = getLocalStorageItem<TransferFormModel>(
```

Comment:
Instead of using `setLocalStorageItem` and `getLocalStorageItem` I suggest to use `useStorageRef`.

Reply:
I could not get this to work correctly
https://github.com/aeternity/superhero-wallet/pull/2322#discussion_r1322945599